core/core.go

```diff
@@ -206,12 +207,29 @@ func (gadget Gadget) Run() error {
 				if !exists {
 					route = gadget.Router.DefaultMentionRoute
 				}
+
 				if !gadget.Router.Can(currentUser, route.Permissions) {
 					log.Warn().Str("user", currentUser.Uuid).Str("route", route.Name).Msg("Permission failure")
 					route = gadget.Router.DeniedMentionRoute
 				}
+
 				log.Debug().Str("user", currentUser.Uuid).Str("route", route.Name).Msg(trimmedMessage)
-				// run the code in its own goroutine so we can return a result to Slack quickly
+
+				go route.Execute(*api, gadget.Router, *ev, trimmedMessage)
+			case *slackevents.MessageEvent:
+				gadget.Router.DbConnection.FirstOrCreate(&currentUser, models.User{Uuid: ev.User})
+
+				if err != nil {
+					w.WriteHeader(http.StatusInternalServerError)
+					return
+				}
+
+				trimmedMessage := stripBotMention(ev.Text, myUuid)
+				route, exists := gadget.Router.FindChannelMessageRouteByMessage(trimmedMessage)
+				if !exists {
+					route = gadget.Router.DefaultChannelMessageRoute
+				}
+
 				go route.Execute(*api, gadget.Router, *ev, trimmedMessage)
```

Comment:
I think for this, we should be careful. I'm not sure what you're planning on making `DefaultChannelMessageRoute`, but it should probably be very different than the default mention route which returns back a "I don't know what to do with this" type response. Where is your default defined (or is it yet)?

Reply:
FWIW, my default for mentions is in the `plugins/fallback` plugin. If we're going to have a default and make it required, we should probably provide _something_ sane. Thoughts?

Reply:
I wasn't intending this to necessarily do anything, to simply be a no-op by default and allow it to be set (with caution) downstream. We _could_ do something like check a user's acceptance of terms before allowing them to post to a channel or to have the bot be friendly to new users. Though this could easily be implemented with a greedy handler.

I agree that unless we have a clear use case for it we probably shouldn't implement it at the moment. I'll remove this and if we have a need for it later we can use it.